lib/jackal-kitchen/formatter/adjudicate.rb

```diff
@@ -112,6 +79,42 @@ def spec_metadata(data, format)
             :total_runtime => duration
           )
         end
+
+        private
+
+        # For each test category, populate a results array that indicates
+        #   any test failures
+        #
+        # @param payload [Smash] payload data with test example info
+        # @return [Hash] eg: { reasons: [teapot: ['I was born to fail'], chefspec: [], ...]}
+        def populate_reasons_for_failure(payload)
+          reasons = {}
+
+          [:chefspec, :serverspec, :teapot].each do |type|
+            reasons[type] = []
+            examples = payload[:data][:kitchen][:test_output][type][:examples] || []
+            examples.select { |h| h[:status] == 'failed' }.each do |h|
+              reasons[type] << h[:description]
+            end
+
+            cond = (type == :teapot &&
```

Comment:
we can do this for all types, as they all support` [:total_runtime][:threshold_exceeded]` (it was just misleading in the old code because it was implemented as an example